src/stagemodel/node_model.py

```diff
@@ -98,19 +99,32 @@ def create_design_mat(self, data: MRData = None) -> np.ndarray:
             np.ndarray: Design matrix.
         """
         data = self.data if data is None else data
+        assert isinstance(data, MRData)
         return np.hstack([cov_model.create_design_mat(data)[0]
                           for cov_model in self.cov_models])
 
+    def create_design_mat_from_xarray(self, covs: List[xr.DataArray]) -> np.ndarray:
+        var_coord = "variable" if len(covs) == 1 else "variables"
```

Comment:
Actually I think this is just always "variable". Can you tell me when you see "variables" pop up?